README

В процессе написание кода программе нехватало выделенной памяти. На момент последнего запкуска на моей машине ошибки были устранены, надеюсь, у Вас всё будет работать :)
Для повторения представленных результатов достаточно запустить последовательное исполнение всех ячеек.

# Финальное задание

## Подход №1. Градиентый бустинг. 

### 1. Считывание таблицы с признаками. Удаление признаков, связанных с итогами матча.

In [1]:
import pandas
raw_features = pandas.read_csv('./features.csv', index_col='match_id')

print('initial shape: '+ str(raw_features.shape))

features = raw_features.drop(['duration', 'radiant_win', 'tower_status_radiant', 'tower_status_dire', 
                              'barracks_status_radiant', 'barracks_status_dire'], axis=1)

print('new shape: '+ str(features.shape))
features.head()

initial shape: (97230, 108)
new shape: (97230, 102)


start_time  lobby_type  r1_hero  r1_level  r1_xp  r1_gold  r1_lh  \
match_id                                                                     
0         1430198770           7       11         5   2098     1489     20   
1         1430220345           0       42         4   1188     1033      9   
2         1430227081           7       33         4   1319     1270     22   
3         1430263531           1       29         4   1779     1056     14   
4         1430282290           7       13         4   1431     1090      8   

          r1_kills  r1_deaths  r1_items          ...           \
match_id                                         ...            
0                0          0         7          ...            
1                0          1        12          ...            
2                0          0        12          ...            
3                0          0         5          ...            
4                1          0         8          ...            

          radiant_ward_sentry_count  radiant_first_ward_time  \
match_id                                                       
0                                 0                     35.0   
1                                 0                    -20.0   
2                                 1                    -39.0   
3                                 0                    -30.0   
4                                 0                     46.0   

          dire_bottle_time  dire_courier_time  dire_flying_courier_time  \
match_id                                                                  
0                    103.0              -84.0                     221.0   
1                    149.0              -84.0                     195.0   
2                     45.0              -77.0                     221.0   
3                    124.0              -80.0                     184.0   
4                    182.0              -80.0                     225.0   

          dire_tpscroll_count  dire_boots_count  dire_ward_observer_count  \
match_id                                                                    
0                           3                 4                         2   
1                           5                 4                         3   
2                           3                 4                         3   
3                           0                 4                         2   
4                           6                 3                         3   

          dire_ward_sentry_count  dire_first_ward_time  
match_id                                                
0                              2                 -52.0  
1                              1                  -5.0  
2                              1                  13.0  
3                              0                  27.0  
4                              0                 -16.0  

[5 rows x 102 columns]

### 2. Проверка выборки на наличие пропусков. Вывод названий признаков, имеющих пропуски. Представление обоснование пропуска значений этих признаков для любых 2-х из них.

In [2]:
nan_counts = features.count()
target_nan_counts = nan_counts.where(nan_counts < features.shape[0]).dropna()
print(target_nan_counts.shape[0])
target_nan_counts

12


first_blood_time               77677.0
first_blood_team               77677.0
first_blood_player1            77677.0
first_blood_player2            53243.0
radiant_bottle_time            81539.0
radiant_courier_time           96538.0
radiant_flying_courier_time    69751.0
radiant_first_ward_time        95394.0
dire_bottle_time               81087.0
dire_courier_time              96554.0
dire_flying_courier_time       71132.0
dire_first_ward_time           95404.0
dtype: float64

Больше всего пропусков наблюдается в данных, описывающих первое убийство героя одной из команд. Очевидно, не все игроки готовы в первые пять минут рисковать и нападать на противника. Атаки, особенно успешные, в основном случаются у сыгранных команд, или команд состоящих из опытных игроков, способных на хорошее взаимодействие.

### 3. Замена пропусков на 0. 

In [3]:
features = features.fillna(0)
features.count().where(nan_counts < features.shape[0]).dropna()

first_blood_time               97230.0
first_blood_team               97230.0
first_blood_player1            97230.0
first_blood_player2            97230.0
radiant_bottle_time            97230.0
radiant_courier_time           97230.0
radiant_flying_courier_time    97230.0
radiant_first_ward_time        97230.0
dire_bottle_time               97230.0
dire_courier_time              97230.0
dire_flying_courier_time       97230.0
dire_first_ward_time           97230.0
dtype: float64

### 4. Целевую переменную содержит столбец с названием radiant_win: 1, если победила команда Radiant, 0 — иначе

### 5. Обучение градиентного бустинга.

In [4]:
from sklearn.model_selection import KFold, cross_val_score
from sklearn.ensemble import GradientBoostingClassifier
import datetime

time_of_training_of_30_trees = 0

tree_numbers = [5, 10, 15, 20, 25, 30, 50, 80]
kf = KFold(n_splits=5, random_state=241, shuffle=True)

x = features
y = raw_features['radiant_win']

start_time = datetime.datetime.now()

for i in tree_numbers:
    if i == 30:
        time_of_training_of_30_trees = datetime.datetime.now()
        
    
    gb = GradientBoostingClassifier(n_estimators=i, random_state=241)
    scr = cross_val_score(gb, x, y, scoring='roc_auc', cv=kf)
    score= sum(scr) / len(scr)
    print('{}: {}'.format(i, score))
    
    if i == 30:
        time_of_training_of_30_trees=datetime.datetime.now()-time_of_training_of_30_trees


print('Total training time:', datetime.datetime.now() - start_time)
print('30 trees training time:', time_of_training_of_30_trees)


%reset_selective -f kf, gb, scr

5: 0.6357376430919737
10: 0.6643877206345741
15: 0.6757420237293614
20: 0.6828535735340823
25: 0.6868481585275606
30: 0.6894962060591201
50: 0.6974548316948366
80: 0.70381673531275
Total training time: 0:08:15.320589
30 trees training time: 0:01:02.503182


Классификаторы настраивались достаточно долго. общее время настройки ~ 8 минут 10 секунд, а градиентный бустинг с 30-ю деревьями ~ 1 минуту 2 секунды. Качество классификации на 30-ти деревьях достигла 0.6895, а на 80-ти - 0.7038. Можно сделать вывод, что имеет смысл использовать больше 30 деревьев в градиентном бустинге, однако при увеличении количества деревьев производительность будет падать существенно. Для ускорения обучение можно проводить обучение по случайной подвыборке множества. Можно ограничить глубину деревьев, однако параметр нужно тщательно подбирать, чтобы это не приводило к падению качества обучения.

## Подход №2. Логистическая регрессия.

### 1. Оценка качества логистической регрессии.

In [5]:
import numpy as np
import warnings
import datetime

from sklearn.model_selection import KFold, cross_val_score
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression

warnings.filterwarnings("ignore")

scaler = StandardScaler()
categorical_scaled_features = scaler.fit_transform(features)

x = categorical_scaled_features
y = raw_features['radiant_win']

C_values = [0.001, 0.005, 0.01, 0.05, 0.1, 0.3, 0.5, 0.7, 0.9, 1, 2, 5, 7, 10, 15, 20, 30]

start_time = datetime.datetime.now()

kf = KFold(n_splits=5, random_state=241, shuffle=True)
for i in C_values:
    lr = LogisticRegression(penalty='l2',C=i)
    scr = cross_val_score(lr, x, y, scoring='roc_auc', cv=kf)
    score = sum(scr) / len(scr)
    print('{}: {}'.format(i, score))


print('Total training time:', datetime.datetime.now() - start_time)


%reset_selective -f kf, lr, scaler, categorical_scaled_features, scr

0.001: 0.7161802463683579
0.005: 0.7163566911760233
0.01: 0.716341462186996
0.05: 0.7163142089233235
0.1: 0.7163100836533355
0.3: 0.7163078591526911
0.5: 0.7163074013893939
0.7: 0.7163070095008914
0.9: 0.7163066324079816
1: 0.716306583645544
2: 0.7163064649978694
5: 0.7163064691888688
7: 0.7163063420811311
10: 0.7163063399602339
15: 0.7163063145103399
20: 0.7163062975560178
30: 0.7163062594232421
Total training time: 0:04:25.449988


Наилучшее качество логистической регрессии над всеми исходными признаками ~ 0.71634. Оно лучше, чем качество градиентного бустинга на 80-ти деревьях. По сравнению с 30-ю деревьями разница существенна. Дело в том, что логистическая регрессия содержит гораздо меньше весов, которые необходимо настроить, её структура проще, чем у решающих деревьев. Поэтому, она настраивается гораздо быстрее и её качество в начале обучения как правило достаточно близко к оптимуму. Из-за существенной разницы в количестве настраиваемых параметров логистическая регрессия работает сравнимо быстрее градиентного бустинга.

### 2. Оценка логистической регрессии без использования категориальных признаков в обучении.

In [6]:
import numpy as np
import warnings
import datetime

from sklearn.model_selection import KFold, cross_val_score
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression

warnings.filterwarnings("ignore")

scaler = StandardScaler()
cleaned_features = features.drop(['lobby_type', 'r1_hero', 'r2_hero', 'r3_hero', \
                        'r4_hero', 'r5_hero', 'd1_hero', 'd2_hero', 'd3_hero', 'd4_hero', 'd5_hero'], axis=1)
scaled_cleaned_features = scaler.fit_transform(cleaned_features)

x = scaled_cleaned_features
y = raw_features['radiant_win']

print(scaled_cleaned_features.shape)

C_values = [0.001, 0.005, 0.01, 0.05, 0.1, 0.3, 0.5, 0.7, 0.9, 1, 2, 5, 7, 10, 15, 20, 30]

start_time = datetime.datetime.now()

kf = KFold(n_splits=5, random_state=241, shuffle=True)
for i in C_values:
    lr = LogisticRegression(penalty='l2',C=i)
    scr = cross_val_score(lr, x, y, scoring='roc_auc', cv=kf)
    score = sum(scr) / len(scr)
    print('{}: {}'.format(i, score))


print('Total training time:', datetime.datetime.now() - start_time)


%reset_selective -f kf, lr, scaler, cleaned_features, scaled_cleaned_features, scr

(97230, 91)
0.001: 0.7162355910206267
0.005: 0.7164130435225277
0.01: 0.7164009506527343
0.05: 0.7163773357189422
0.1: 0.7163737844721111
0.3: 0.7163710492777403
0.5: 0.7163706000071687
0.7: 0.7163707356064428
0.9: 0.7163706488113248
1: 0.7163707526581122
2: 0.7163706806053447
5: 0.7163705132143667
7: 0.7163704496394463
10: 0.7163704793048005
15: 0.7163704475465819
20: 0.7163704306102077
30: 0.7163704475554
Total training time: 0:04:09.560851


2. Наилучшее качество после удаления из выборки категориальных признаков ~ 0,7164. Оно чуть лучше по сравнению с предыдущем примером. Это связано с тем, что категориальный признак, будучи нормированным, является малоинформативным для классификатора, может приводить к ухудшению результата

### 3. Количество уникальных героев в игре.

In [7]:
names_to_insert = ['r', 'd']

set_of_heroes = set()
for i in range(1, 6):
    set_of_heroes = set_of_heroes.union(set(raw_features['r'+str(i)+'_hero'].unique()))
    set_of_heroes = set_of_heroes.union(set(raw_features['d'+str(i)+'_hero'].unique()))
print('max = {}'.format(max(set_of_heroes)), 'len = {}'.format(len(set_of_heroes)), sep='\n')

%reset_selective -f set_of_heroes

max = 112
len = 108


Ответ: 112

### 4. Кодирование категориальных признаков "мешком слов".

In [8]:
import numpy as np
import warnings

warnings.filterwarnings("ignore")

X_pick = np.zeros((features.shape[0], 112))

for i, match_id in enumerate(features.index):
    for p in range(5):
        X_pick[i, features.ix[match_id, 'r%d_hero' % (p+1)]-1] = 1
        X_pick[i, features.ix[match_id, 'd%d_hero' % (p+1)]-1] = -1

print(X_pick.shape)

(97230, 112)


### 5. Кроссвалидация признаков на новой выборке.

In [9]:
import numpy as np
import warnings
import datetime

from sklearn.model_selection import KFold, cross_val_score
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression

warnings.filterwarnings("ignore")

scaler = StandardScaler()
cleaned_features = features.drop(['lobby_type', 'r1_hero', 'r2_hero', 'r3_hero', \
                        'r4_hero', 'r5_hero', 'd1_hero', 'd2_hero', 'd3_hero', 'd4_hero', 'd5_hero'], axis=1)
scaled_cleaned_features = scaler.fit_transform(cleaned_features)

print(scaled_cleaned_features.shape)

scaled_cleaned_features_with_bow = np.concatenate((scaled_cleaned_features, X_pick), axis=1)

%reset_selective -f X_pick

print(scaled_cleaned_features_with_bow.shape)

x = scaled_cleaned_features_with_bow
y = raw_features['radiant_win']

C_values = [0.001, 0.005, 0.01, 0.05, 0.1, 0.3, 0.5, 0.7, 0.9, 1, 2, 5, 7, 10, 15, 20, 30]

start_time = datetime.datetime.now()

kf = KFold(n_splits=5, random_state=241, shuffle=True)
for i in C_values:
    lr = LogisticRegression(penalty='l2',C=i)
    scr = cross_val_score(lr, x, y, scoring='roc_auc', cv=kf)
    score = sum(scr) / len(scr)
    print('{}: {}'.format(i, score))


print('Total training time:', datetime.datetime.now() - start_time)


%reset_selective -f kf, lr, scaler, cleaned_features, scaled_cleaned_features, scr

(97230, 91)
(97230, 203)
0.001: 0.7462962333706958
0.005: 0.7512162759469865
0.01: 0.7517359377885027
0.05: 0.7519446302262013
0.1: 0.7519375046034006
0.3: 0.7519250173114229
0.5: 0.7519220537466547
0.7: 0.7519206765413137
0.9: 0.7519197080706709
1: 0.751919733264219
2: 0.7519181722092281
5: 0.7519173055919961
7: 0.7519172865150461
10: 0.751917180623139
15: 0.7519170450804328
20: 0.751917165796959
30: 0.751917237901924
Total training time: 0:07:25.368594


Наилучшее качество ~ 0.75194. Это лучше, чем достигнутые ранее результаты. Объясняется это тем, что теперь логистическая регрессия может использовать преобразованные категориальные признаки максимально информативно.

Лучшего качества модель достигает при С=0,05. Логистическая регрессия с этим параметром будет использоваться на тестовой выборке.

### 6. Построение предсказаний вероятностей победы команды Radiant для тестовой выборки с помощью логистической регрессии с преобразованием категориальных признаков.

In [10]:
import pandas
test_features = pandas.read_csv('./features_test.csv', index_col='match_id')

print('shape: '+ str(test_features.shape))

test_features.head()

shape: (17177, 102)


start_time  lobby_type  r1_hero  r1_level  r1_xp  r1_gold  r1_lh  \
match_id                                                                     
6         1430287923           0       93         4   1103     1089      8   
7         1430293357           1       20         2    556      570      1   
10        1430301774           1      112         2    751      808      1   
13        1430323933           1       27         3    708      903      1   
16        1430331112           1       39         4   1259      661      4   

          r1_kills  r1_deaths  r1_items          ...           \
match_id                                         ...            
6                0          1         9          ...            
7                0          0         9          ...            
10               0          0        13          ...            
13               1          1        11          ...            
16               0          0         9          ...            

          radiant_ward_sentry_count  radiant_first_ward_time  \
match_id                                                       
6                                 0                     12.0   
7                                 2                    -29.0   
10                                1                    -22.0   
13                                2                    -49.0   
16                                0                     36.0   

          dire_bottle_time  dire_courier_time  dire_flying_courier_time  \
match_id                                                                  
6                    247.0              -86.0                     272.0   
7                    168.0              -54.0                       NaN   
10                    46.0              -87.0                     186.0   
13                    30.0              -89.0                     210.0   
16                   180.0              -86.0                     180.0   

          dire_tpscroll_count  dire_boots_count  dire_ward_observer_count  \
match_id                                                                    
6                           3                 4                         2   
7                           3                 2                         2   
10                          1                 3                         3   
13                          3                 4                         2   
16                          1                 3                         2   

          dire_ward_sentry_count  dire_first_ward_time  
match_id                                                
6                              0                 118.0  
7                              1                  16.0  
10                             0                 -34.0  
13                             1                 -26.0  
16                             1                 -33.0  

[5 rows x 102 columns]

In [11]:
nan_counts = test_features.count()
target_nan_counts = nan_counts.where(nan_counts < test_features.shape[0]).dropna()
print(target_nan_counts.shape[0])
target_nan_counts

12


first_blood_time               13625.0
first_blood_team               13625.0
first_blood_player1            13625.0
first_blood_player2             9411.0
radiant_bottle_time            14282.0
radiant_courier_time           17050.0
radiant_flying_courier_time    12292.0
radiant_first_ward_time        16847.0
dire_bottle_time               14335.0
dire_courier_time              17047.0
dire_flying_courier_time       12653.0
dire_first_ward_time           16914.0
dtype: float64

In [12]:
test_features = test_features.fillna(0)
test_features.count().where(nan_counts < test_features.shape[0]).dropna()

first_blood_time               17177.0
first_blood_team               17177.0
first_blood_player1            17177.0
first_blood_player2            17177.0
radiant_bottle_time            17177.0
radiant_courier_time           17177.0
radiant_flying_courier_time    17177.0
radiant_first_ward_time        17177.0
dire_bottle_time               17177.0
dire_courier_time              17177.0
dire_flying_courier_time       17177.0
dire_first_ward_time           17177.0
dtype: float64

In [13]:
names_to_insert = ['r', 'd']

set_of_heroes = set()
for i in range(1, 6):
    set_of_heroes = set_of_heroes.union(set(test_features['r'+str(i)+'_hero'].unique()))
    set_of_heroes = set_of_heroes.union(set(test_features['d'+str(i)+'_hero'].unique()))
print('max = {}'.format(max(set_of_heroes)), 'len = {}'.format(len(set_of_heroes)), sep='\n')

max = 112
len = 108


In [14]:
X_pick = np.zeros((test_features.shape[0], 112))

for i, match_id in enumerate(test_features.index):
    for p in range(5):
        X_pick[i, test_features.ix[match_id, 'r%d_hero' % (p+1)]-1] = 1
        X_pick[i, test_features.ix[match_id, 'd%d_hero' % (p+1)]-1] = -1

print(X_pick.shape)

(17177, 112)


In [15]:
import numpy as np
import warnings
import datetime

from sklearn.model_selection import KFold, cross_val_score
from sklearn.preprocessing import StandardScaler
from sklearn.linear_model import LogisticRegression

scaler = StandardScaler()
cleaned_test_features = test_features.drop(['lobby_type', 'r1_hero', 'r2_hero', 'r3_hero', \
                        'r4_hero', 'r5_hero', 'd1_hero', 'd2_hero', 'd3_hero', 'd4_hero', 'd5_hero'], axis=1)
scaled_cleaned_test_features = scaler.fit_transform(cleaned_test_features)

print(scaled_cleaned_features.shape)

scaled_cleaned_test_features_with_bow = np.concatenate((scaled_cleaned_test_features, X_pick), axis=1)

print(scaled_cleaned_test_features_with_bow.shape)

(97230, 91)
(17177, 203)


In [16]:
warnings.filterwarnings("ignore")

x = scaled_cleaned_features_with_bow
y = raw_features['radiant_win']

print(x.shape, y.shape, sep='\n')


x_test = scaled_cleaned_features_with_bow

lr = LogisticRegression(penalty='l2',C=i)
lr.fit(x, y)

pred = lr.predict_proba(x_test)

pred

%reset_selective -f kf, lr, scaler, cleaned_test_features, scaled_cleaned_test_features, scr

(97230, 203)
(97230,)


Предсказания:

In [17]:
target_pred = [x[1] for x in pred]
for x in target_pred:
    print(x)

0.6714344983556745
0.7368763634311486
0.6038308010372452
0.2320375062109482
0.278889375275172
0.1708929041135347
0.29171381061234053
0.29732144949596145
0.22574624994241432
0.38702309785322947
0.9212424667330473
0.1534091079642786
0.1781201271606578
0.757662320559931
0.0773245306211463
0.6792352265339636
0.47408200948872997
0.4507915854584413
0.3458091885592436
0.5249918888611297
0.2762183223059895
0.900141150297187
0.8459472122973344
0.14502141359249285
0.5563248235397974
0.8036373625529537
0.615657401238632
0.7391435374111278
0.6010161560341363
0.2992192699937047
0.11664864432858486
0.4576864267407968
0.9407113886210606
0.453050710930943
0.16791725714564773
0.08144531243286438
0.8179816228204377
0.3583156854346834
0.3016304050546374
0.43765549291178885
0.8383298051319671
0.18826946468191197
0.7683596844273038
0.29999205018566316
0.11249949461602612
0.2777988816784539
0.5234275319857189
0.4755738086742744
0.49314819639691854
0.12128474744461029
0.4103082545243292
0.33034074873471064
0

0.10490091750697564
0.4532821935967241
0.23637095030415128
0.6857023527809384
0.46345393851067895
0.24834731595960435
0.44916356279936764
0.13213248301807307
0.7332270546857367
0.0798281148617621
0.13477307669678412
0.8170872224380012
0.23846858489913228
0.3689911755715115
0.666978516877291
0.7949460723075971
0.36936094044424506
0.36913615758680474
0.9452318142608169
0.22494331865011546
0.4349541320979598
0.06449383846266368
0.13118471360924794
0.6607544271663613
0.8363212730377528
0.5463878937517579
0.6345790103236504
0.516838368664151
0.4620311872087981
0.3009475769640904
0.464214526261513
0.5474695102015198
0.5737181828899263
0.49824623767639825
0.46887959216080916
0.6286467578262809
0.5311292273262634
0.4764359884634827
0.4180322649213218
0.7555411812464992
0.3623413427557656
0.2406079899152785
0.18903886759181426
0.39236585752586856
0.14636750496192252
0.3188584768735876
0.3867037266502266
0.673847891314961
0.6879821872610952
0.360592297112715
0.6074082706745868
0.7334543863404568

0.7108901865180901
0.9292842827420331
0.804738876163615
0.6806469876314653
0.6609364612067125
0.3257606988642287
0.6367294001672911
0.736143818023368
0.7965288271317623
0.5256114682717138
0.47212064714015856
0.7629692091905391
0.3202459643107093
0.7170887997461807
0.6461408323339527
0.9765251361985454
0.12360879666214665
0.8694262648366153
0.4697079466537662
0.2865772726770247
0.1527810800210883
0.5978323381951743
0.470608994001026
0.583326570797388
0.5091962515781783
0.9182954120272631
0.3971665467563686
0.642435912136262
0.08410383309150564
0.08156286631353941
0.7859473932582208
0.20243884315876423
0.3145941539670811
0.8764860787283207
0.3150766600015355
0.47613495013264284
0.5923893598160705
0.8212911376976828
0.868542384086385
0.4393474837212566
0.9049770747335135
0.42369970365750936
0.5780475885328512
0.4246219339897425
0.4961565405401314
0.8122804021623538
0.508397201648175
0.4827762894244679
0.8543590660559696
0.6500487922146811
0.5610968828240447
0.6488264581542366
0.6608445790

0.32287390665163546
0.5836246380573402
0.8264505369107259
0.638963584466153
0.21425304070275716
0.7149329805082165
0.22493360912122676
0.15644567267535858
0.5988032095832694
0.658069058857291
0.5640890978776462
0.768449277061874
0.36596658930496506
0.16083109856919914
0.4909126431104298
0.7949002629340397
0.6216407767128169
0.21425576121830892
0.6366669834172083
0.2683019908236647
0.639591553050854
0.6707238129667039
0.18332548205051674
0.5430800095011257
0.2553094470540794
0.5935212433628239
0.6857100590283485
0.35220468997274673
0.6991364438139447
0.8674767002023386
0.19193769974337865
0.7700304345390622
0.6525664589031138
0.4856745173002416
0.6884401680978135
0.699827844976514
0.7502647952268987
0.76624598106479
0.12519788989171457
0.5201293762318061
0.23913294447396738
0.46682442766808885
0.596695684584207
0.41005062158559674
0.8911391062245451
0.7905543404866636
0.7594875058494772
0.4937039567413094
0.5736930710712836
0.6423733494852079
0.4171899587089401
0.3084764235156731
0.2682

0.5148847685187898
0.5118862188301941
0.7188930864464053
0.8714136381862745
0.8929484180686984
0.4148432195297646
0.06924389395076787
0.7889172841980772
0.6738450254367999
0.5838528090468016
0.8020886603529732
0.290021371612642
0.1620634843043778
0.05892584367286231
0.9356754009574275
0.2511977310468375
0.5462922352623739
0.6406652695384449
0.5032232948215614
0.5177538944027026
0.5190287814718303
0.2209216898174558
0.5120013461868418
0.09727205927793234
0.2968578042437737
0.3031723225255121
0.19693314351278396
0.570312669825779
0.3713416942484158
0.34958706679811447
0.5552891875597811
0.4280197997699174
0.6214875135086986
0.38519063207968873
0.41485232359786195
0.4656438248190281
0.2796660080313805
0.3073352572341398
0.4032814973114358
0.3980620214070122
0.8100652734099937
0.25684469885772426
0.8984648716297208
0.8006293088305414
0.74569068912383
0.4492690537543894
0.22322198047745268
0.22679970671550678
0.5245395792232691
0.16615222784315384
0.4742271793199179
0.22881895059453503
0.88

0.6235997202383753
0.7313364615502453
0.8816290283091064
0.5804087172713805
0.4595192145710339
0.8158308956155369
0.9789278387412063
0.6610019596662822
0.9895006582984337
0.6011463331620671
0.337342419703136
0.4607668518798773
0.7449673157676465
0.16833877856043603
0.5446303871869727
0.3892166724326599
0.5277567654115751
0.6217735912730777
0.7267708571586384
0.5415592696250502
0.710034971788037
0.22628085987782814
0.6231610803730072
0.37790320950511197
0.303474327926449
0.08013095798916237
0.6288302998787424
0.2390497752273188
0.27863184179676037
0.6825471239001676
0.4062045500583716
0.21597259141775083
0.8219551742977113
0.26215689160301436
0.45130920822234727
0.42699975523055084
0.36796489708768493
0.4319575404593786
0.9188771533043876
0.4791220752886428
0.7762542730130251
0.41385676903474927
0.6126945130997784
0.8281956531662306
0.2912402025152906
0.5731293187140472
0.11719024084154456
0.8128643922047281
0.5452167981069155
0.14889344918668976
0.7071600938427023
0.8613879217982233
0.

0.13966397764425997
0.894981004952403
0.5044359116810336
0.7024761267253292
0.13145301865666076
0.492906081909739
0.7830916163559967
0.6078120454224694
0.4637528149209078
0.9103266091001289
0.4121898377496688
0.4790329468844455
0.1881302317133455
0.4189674723252519
0.8375425877448802
0.7900650292517247
0.6521823363384743
0.7829462216372796
0.665061338735982
0.7614860304816021
0.4035528595984417
0.09999490257838343
0.9059774710601747
0.8299122400077091
0.6476106031752131
0.5702441921736943
0.3611700537319816
0.29877419194410715
0.08467590977139851
0.3100249354903927
0.22262065156916108
0.7585315008317699
0.43742009265504656
0.4252020513359044
0.6523131188998619
0.5710020959654257
0.025703064690293893
0.7539945909295909
0.5500028072526836
0.3465121808264663
0.22300527547058865
0.5277683673345074
0.9298221873350544
0.6400977781609053
0.5090209378595185
0.1781325599812
0.8119940221738295
0.6679789171248645
0.7317099226946636
0.6816741387683207
0.012005655653155901
0.7429143761708703
0.6745

0.8384307459568344
0.3259102645963147
0.8797125972931222
0.574619110226599
0.49923424916042236
0.7404502517990951
0.39053049770562304
0.49358477247796434
0.08886481551566695
0.38878764466859267
0.3729222313340829
0.3920881084497632
0.7058379593508612
0.9561396406780093
0.23309504913797183
0.6873983142616134
0.15136708634897222
0.3731065786003654
0.6872587612019021
0.1421549306632058
0.4245431649791331
0.7155318822295597
0.011162577017774662
0.8565182087486359
0.3315092637248231
0.6416740611855443
0.5493392095952492
0.6180896361696102
0.8418588827978656
0.3497856946044408
0.6853257706386426
0.20504541595462952
0.48628351823166055
0.5546639590456952
0.25660501549027076
0.6698746278763555
0.17677520442912903
0.7903410307896348
0.499565821795199
0.7765704386179965
0.5115556191844777
0.7257420347599095
0.661400553791508
0.5342209957184196
0.30836756535679105
0.08718021479641351
0.8650454709418317
0.4405883311572886
0.5762947404011685
0.6040484613669846
0.5590033127907834
0.8878427466637763


0.5460131713709556
0.47872189756941824
0.4185126895280257
0.2014609472162942
0.3780198865567478
0.49238647527591156
0.06592602788970799
0.35464710110956404
0.5970476370219666
0.3644087355705939
0.5474146114035573
0.2286943441230665
0.6239129344182274
0.9522430503327266
0.6832159242057021
0.22190721780842348
0.607193493805653
0.8430723445917835
0.30226780307559453
0.3872587208798561
0.9595500728832523
0.6669415387416191
0.7969477049386061
0.28103958258968254
0.3112067849533984
0.7596068264242748
0.5589254901579795
0.31417907736398015
0.3602559518396966
0.03432920836366575
0.4746016517763014
0.2307222199106501
0.49639677777169006
0.3076422746029917
0.5155791455987369
0.6068124011897046
0.5510755063385451
0.03780617354597114
0.2522579172551349
0.5947306668823299
0.6017469474656089
0.564254407361603
0.3111355031540379
0.3236972199520607
0.8438980549167163
0.7016392598279886
0.8153437970027023
0.9527541320003385
0.8865530304280023
0.21778438850860327
0.238072344567835
0.8032306687724877
0.3

0.803287323200993
0.5720813708802509
0.3544814714905518
0.7188450082395181
0.3164785172639884
0.6111652500931407
0.22343231886883044
0.32898669177907053
0.5002538378623725
0.2925945531427917
0.5919855991295007
0.14233881019991063
0.9286050455374495
0.20015190944386918
0.08149114651016058
0.7826619596469597
0.7191247878969116
0.4065383545322949
0.408866951462775
0.3013719330153585
0.3271564083027679
0.8906055592077937
0.3941826446171659
0.3833328276919536
0.931475543862837
0.5828411305107033
0.2494798020035097
0.7417509356836978
0.11821732215286568
0.617863923971521
0.304922216904051
0.7372045268878454
0.44774087936259443
0.5250126431331832
0.5081918283507685
0.1612997428816711
0.9071927959986076
0.3257784233188421
0.8903312115070662
0.7205812668636931
0.852155042360151
0.8987202266043836
0.551591375080114
0.3983675508333062
0.5658314899935473
0.3426385308399707
0.9704822783960234
0.3057578445509884
0.44970843530543975
0.8986578459276776
0.5204391321670933
0.28575310772773765
0.18864776

0.7069421229361729
0.385057817946189
0.532020489275666
0.590918343864474
0.08448150543875792
0.5928277653664055
0.0740287849794274
0.19141901622817697
0.5168146243928358
0.5959864964464306
0.7961609403838764
0.7224817500796473
0.7152985310349337
0.47497359923346943
0.7749729929626853
0.3015647828441108
0.649573262338456
0.3236388673439997
0.6243784758445724
0.39743807951953497
0.7341256808025284
0.44846311954602475
0.3474704959274435
0.30616856027307504
0.6023168742354279
0.22656622220610137
0.2898060331248899
0.6901329883253017
0.7410875483984574
0.27500517458930646
0.33533280007055205
0.7153696296296682
0.4066110381418213
0.30290677278777955
0.1438191930717252
0.27110908862777694
0.5513703349903724
0.7040571862984204
0.5886352840388663
0.5904221367073632
0.5585320580921109
0.568616770092707
0.4380000611952094
0.5608811781914578
0.42890862017348746
0.3529189258403681
0.2527192301291898
0.52759744884091
0.10031686394362024
0.5953482034004681
0.1063331366749847
0.5520314410510729
0.2809

0.6958484681521769
0.4644485764248529
0.4059042024350146
0.439796668535054
0.7134080842596631
0.7812814949259583
0.915088097536789
0.8665248678252064
0.3772260375931342
0.866706918563566
0.08685970871707951
0.44715517495410106
0.5205850885033497
0.2930882222935291
0.5726018986106397
0.21672498949615512
0.5368986233865569
0.464387888266134
0.6456871873724533
0.7444435236260476
0.7229181868967075
0.679028547148918
0.35734609073105866
0.7560393613398756
0.2756367798205152
0.685638501302023
0.4329564359317018
0.8348005982313315
0.24771874165633548
0.2145792575965541
0.8716214251970336
0.7330445863749253
0.6506562767804189
0.6921208067069735
0.2088431056044985
0.14077444233569286
0.4667798563292168
0.4641390506194613
0.8959041405784669
0.3163592944968694
0.5785473264859807
0.4129386596041757
0.9099283986746829
0.6815453909610434
0.5983468429936066
0.7264292298213076
0.2743286633810958
0.1645715614383274
0.45060988845002475
0.6715101581186015
0.4824963701424076
0.7238177355145651
0.503266041

0.43934056891239737
0.49231584132873496
0.47877246103983073
0.25536722174243365
0.7140195440506014
0.5729552260705145
0.3436942619693822
0.8130213873962299
0.6467938206103494
0.5097300965085299
0.5173862277100181
0.5688077041556385
0.879970432532853
0.22178593794116228
0.440492044595354
0.5442726831086239
0.4212740865061466
0.37469034377161875
0.451367426603683
0.3083597507347711
0.48156075940901283
0.16424463441562095
0.6304515635339651
0.5992400942458814
0.7593116552507244
0.5332954893771403
0.42874602004180995
0.6679239570308493
0.6921598861213716
0.23269098199520613
0.7083777569817754
0.6872805371385445
0.5726309696439207
0.3472609749052273
0.2035189339539975
0.19028332638098108
0.36849750069640047
0.312604222998574
0.5399554113217554
0.6781667816689978
0.3383867524941899
0.6125775229050463
0.7919605803463706
0.4060586449631803
0.3808952046951722
0.2362350464138192
0.7529954814167221
0.7496685165797796
0.597000693144524
0.3615566879172455
0.49758062578488477
0.5095381720841218
0.34

0.7888270552454232
0.949161052582732
0.8234424699659898
0.5031702974466494
0.4502289998228072
0.7799238910981225
0.792149167589883
0.6213281532601028
0.13799941278805386
0.5152774061438333
0.7841243324197811
0.06294231596281016
0.8693776512988679
0.6229732679103902
0.5200621609928412
0.7661664627228668
0.2176694053544837
0.2825446446578992
0.3704952229678536
0.4241009941676999
0.35157602241340935
0.5035346596288661
0.37100693792686495
0.6532126828628192
0.21421176172698927
0.36728068630721505
0.14388222332457187
0.21862686114033508
0.28364751420385065
0.8076991411893744
0.800278071443354
0.9006479118883026
0.43662116159765085
0.6036748619807792
0.2995966802362297
0.4964729587584863
0.722358860940291
0.40713750251795305
0.37554931037933253
0.726359837689092
0.09273630775569118
0.5470700642823398
0.4146423387792032
0.4055336692534807
0.697523188537229
0.3633695678461786
0.6909670394519681
0.7062811743560006
0.12123985185902304
0.3652456488690401
0.3021956666477299
0.7989944497700943
0.16

0.03696145005699312
0.5507200529787681
0.6847496209287269
0.8006721413633457
0.19373086756291946
0.5440967508620917
0.14600136718877052
0.42096039553562703
0.522415849912551
0.6548478156890013
0.40925113571874927
0.7090366257498872
0.1523372159328969
0.5263799196647808
0.5996838377468362
0.8757373756273398
0.4827396791549557
0.5482507082337963
0.19184704747569115
0.27014791125015936
0.4551864633357474
0.604217440307952
0.6639822093302269
0.568066118109196
0.3113763525787462
0.4082572517253876
0.5832265401205114
0.49370119761795567
0.2781740670252242
0.5939370883900373
0.22261556744142164
0.4443446256281008
0.6109043835013662
0.9879622633376659
0.46013770050786107
0.5897339902818279
0.349599165959069
0.3132512192439619
0.5554836766444513
0.5393363289813302
0.8799607084892298
0.5877261633790136
0.5693556308471776
0.7072045778822901
0.25190868012820866
0.8536152357297788
0.8092556135195426
0.8429217975830147
0.7699168106621772
0.6484781922703412
0.12274811348139865
0.2060419731520532
0.27

0.7027369929226502
0.8918544351790656
0.5212055118528552
0.8531738837838405
0.6744464481810614
0.5297042687202426
0.280592130752757
0.7837762090653008
0.33323266969929455
0.3457820153804846
0.4531366138971242
0.6738167114027874
0.3645621873297277
0.16281707828793135
0.9199124706968648
0.4518695237052049
0.8008136354445168
0.5678973767161306
0.3263397687467478
0.40619896073401857
0.44434141009184974
0.39804062125715944
0.9400765580536856
0.6298470298263964
0.6045836560359785
0.22909136997633361
0.4059190983243605
0.4470711400802346
0.9400118115239258
0.8887167908276002
0.5881307012966482
0.3931087182451055
0.40972493675170135
0.8774724570759114
0.5077700991354902
0.4408289259426888
0.15537686195941425
0.58452557200327
0.46603429917891814
0.6375088397386555
0.28817342243158867
0.3714924898694838
0.6293256629099399
0.39981886096225233
0.38729970850136836
0.1789545472391417
0.027582316454726436
0.801011191045296
0.18415841891694837
0.9947955680389388
0.613735393910041
0.3872248541318675
0.

0.6149565478711267
0.8332512304190087
0.2043699226330299
0.36605377017039875
0.46294401158593074
0.812131196947948
0.7772655155611767
0.12175111390179902
0.5180098575558169
0.2627873104029161
0.13337876312237518
0.8632813151708894
0.5111935247215292
0.7127045807302199
0.7659211321086857
0.27476531870893534
0.4655462885478389
0.28922314207683575
0.746738896633093
0.2871869621235929
0.7383485086772867
0.4735453300545082
0.5618319980413541
0.7636502940782327
0.7424519382065989
0.4147566319880812
0.3877609987516528
0.7054525867183294
0.6208372474212227
0.8702124203478581
0.23663796799697823
0.7570846119282438
0.8938424387290651
0.7765319772622133
0.3244665119065388
0.6988148124176724
0.3719931681558942
0.31266698020495093
0.2523269443407384
0.46281185672135533
0.4211726680790604
0.6512493214698442
0.39707998566385644
0.3390409554999427
0.6447443677998052
0.760284191059558
0.7867422171574436
0.9372623377724717
0.3918404230458732
0.6074308586837461
0.35702169335614853
0.7853830659234073
0.33

0.8030671076519185
0.9365656388470043
0.4432702773630136
0.38896023342303265
0.3509790869363758
0.49302506124857826
0.27626019389130324
0.39866627899261153
0.2630732739485724
0.24189383068953424
0.418418070314293
0.6722549836261819
0.1681233460603342
0.111536724535073
0.37974406475749467
0.3666366670196767
0.14447998032675305
0.465001081648163
0.41476053016471426
0.42944428599150575
0.6508212562728418
0.33888567450927704
0.27542077456586966
0.56197894252605
0.11121744304327491
0.339432454056026
0.7428230240664596
0.32663459681795837
0.3406324321287402
0.7385342035343971
0.6346641696611212
0.19078026125757802
0.5716341936717029
0.42828072818397417
0.7407122434774169
0.5500110548552455
0.7893158130847158
0.19497500609477056
0.40274516944378824
0.27628914265076127
0.4421987163203539
0.3150741217871221
0.3925032682459805
0.5599960918768601
0.9704967957013172
0.06713333275897106
0.31916842455176964
0.6415152477808911
0.910313512914532
0.6075694577754166
0.8144601611392148
0.3961027416900173

0.41852731556043743
0.42593581889195586
0.7019196014179727
0.6379060180639727
0.6236408017532044
0.9194989553942323
0.5648703754874665
0.6009002604548523
0.566811286425827
0.35784441496489966
0.7208129328851012
0.6293951947377914
0.4447336635051571
0.6085854674640335
0.6956646393271075
0.39381120797360086
0.2794879619320972
0.5156227766529851
0.8022535271916829
0.40806891117766964
0.48739665057228854
0.49636609974313106
0.7090905113272524
0.5129735787361641
0.5503172733449938
0.055307121489636145
0.4982582344124479
0.15421652028753483
0.48431009119427665
0.174410650871531
0.16233458168023115
0.9083912041347181
0.2591366042275235
0.7617582550231575
0.2689636478224575
0.49809738753261373
0.4929195818728742
0.6694718065687395
0.8793512061214113
0.3937952828260289
0.2911024679611456
0.7302852541130824
0.6092227858421122
0.42668754557438204
0.29604470099519997
0.37921526795091065
0.5172844775872796
0.5038484365099699
0.9476727130264395
0.7157075568878634
0.5515032082827519
0.196610125738034

0.2971086739549391
0.4911658134116931
0.6893452027243147
0.6108520214907914
0.5089677485447549
0.4029343850057598
0.06954947565738884
0.49479780741017093
0.6531796810342756
0.5791837614670916
0.47716717181442886
0.3294843807614102
0.8556543484006002
0.5436841923862433
0.09056698112798096
0.7035345331962773
0.6717267447423773
0.6505488143587682
0.378030759104408
0.7030826284212351
0.800169227952179
0.24410745390361582
0.46058792545459604
0.5697494736266823
0.7637011484155396
0.08642723381251968
0.359596192419672
0.7646091943528535
0.32014706814093297
0.745539251594425
0.3553022453335242
0.42414065820555225
0.10997377826295739
0.8841715575473544
0.21847599428761594
0.851108524226951
0.1842757289480273
0.6270859137400324
0.39100560952685187
0.576054760326958
0.4523892078762985
0.6772416140054641
0.17037623217035555
0.8058979464050673
0.13320871824128958
0.6834573603344205
0.45552082681556644
0.1220488740998867
0.5868147504779251
0.8922608111042916
0.5064657837117436
0.6096508222036128
0.6

0.9040225128825606
0.4731603197294251
0.851421362580401
0.654842790745368
0.6756955573617672
0.7273084731576047
0.32056065446465193
0.6681877770690612
0.48188937011232746
0.5926478858625779
0.3422736786163759
0.42695397642251776
0.4662349401324011
0.7329657671225912
0.2837731561034266
0.5057356550152029
0.2870648573928122
0.3415840001878808
0.5819697648704331
0.6934433679795898
0.7491084579228281
0.5861290482938699
0.524063497643924
0.4506336090911233
0.8631782280001651
0.4307373186790113
0.6326238562128387
0.4278977056398998
0.2297325869362991
0.48564191095623405
0.27535489311206807
0.5837860213022453
0.2353359792300268
0.49659217575054293
0.7386122603833053
0.6672473646793656
0.6866182519667294
0.40376009929426077
0.2664902229912801
0.717923989266419
0.6149111104867052
0.3794275431294425
0.7834129511425643
0.5128932956100012
0.10211572466202537
0.03944162330438413
0.763049421904239
0.7829826794234235
0.5453182259752201
0.4318661802247097
0.5297037322957882
0.4413462765543056
0.802304

0.6707272480626778
0.3471887199043359
0.9267925913375654
0.5004750445272749
0.37922443935048356
0.8720650826033004
0.8256351358784999
0.23771664770731632
0.7178339443972273
0.35215477886695895
0.7265038595634584
0.39253046061515007
0.8563882195343154
0.23937232444196246
0.645743647181059
0.44482446642604945
0.3362385347628627
0.6032903045577445
0.9714090608403679
0.596843631980065
0.1664630145276662
0.45301951719063926
0.7174783600724487
0.6883212631613549
0.17496006694770733
0.26429330406270324
0.6354619039769102
0.6697284440471908
0.8014131701729741
0.9740660252037671
0.8953752519577992
0.43587796615098406
0.30810295824587924
0.5618112590128669
0.5055838877996524
0.8784844240281818
0.3238610084536277
0.6052591172751445
0.9427512315466955
0.9777963930183999
0.5308134668758565
0.8190069105217066
0.4419277540843085
0.881983184880693
0.8067068656195747
0.3456494754055849
0.9359329284354905
0.850066014914677
0.6803012356243798
0.7603261598048807
0.5305804476891741
0.5560676886783432
0.513

0.16493551738701723
0.3138238578804913
0.2874997300312184
0.7222928757248892
0.9185035870782771
0.49363736542222036
0.7339330392380677
0.34734064921851826
0.5746056217264212
0.6860725209524359
0.569089655078196
0.4191252405306275
0.5138280784023154
0.5592573902191228
0.9941212815785191
0.693540814309901
0.464444952730591
0.9811876678392257
0.43090337308797927
0.20461942188075663
0.7347093334548587
0.3244836020587612
0.44905409808089847
0.48642998990397596
0.4213452526153536
0.6576069159326833
0.48276803243578925
0.7364530835326167
0.781055571007506
0.36213466606565065
0.2836384540525191
0.7115084726982941
0.6185044100214686
0.8200184723403532
0.7954144479783604
0.35903266078978413
0.9542781490284404
0.1302431407920963
0.31773830491834504
0.3193821875910057
0.4653965308600609
0.17843382591201223
0.5305865112576634
0.9377420270263362
0.9807177035439965
0.7472317515303176
0.41716395906876547
0.4666648914608294
0.5851127940454864
0.9043937098095606
0.3271719858139655
0.6739393071259707
0.7

0.8056593044885193
0.7730449709406073
0.282013184590329
0.10756929367105092
0.37106329890417195
0.651473607804129
0.8579009238941298
0.3278897637983718
0.6569431468899724
0.30219306409520696
0.6456693767829905
0.850325943503129
0.4179539926427975
0.5200454508809036
0.21951154304852352
0.7340112122094496
0.6274254037474196
0.6915131398801775
0.9319954100059961
0.7864264574991674
0.21668107899888936
0.38644995732169485
0.5233056512624881
0.1672096082113172
0.25618878247498833
0.30547373387460425
0.785024960112723
0.5739831204585519
0.39390471653966735
0.6222780651680818
0.21533418720390518
0.5191555915923871
0.7408537561592123
0.17868299079473912
0.17893197819792309
0.38958197880964524
0.6991807655055169
0.6986647246740411
0.3211716353318358
0.8015369683264143
0.5342418962801094
0.5447292232302711
0.23617383435210712
0.6589927890398489
0.3749049805031864
0.8988656652029418
0.8430992398148605
0.667756832645908
0.24651585971466713
0.09006978199280113
0.044978220601694215
0.5129158989461766

0.4642782851439464
0.5709296594708996
0.499342326101041
0.7813186077483071
0.6237003834290192
0.7602983627679142
0.39383118296962255
0.8960394496909787
0.12157949676312127
0.17270631935482206
0.3896826545539744
0.626776912842023
0.8503891894124357
0.19525245670715552
0.29257811319066135
0.5916262093091459
0.5135042763275541
0.26454066973035334
0.4586194298506782
0.1850441282768484
0.6734690816659301
0.7774660871401923
0.6197494910378202
0.5286562769826881
0.5167514019136995
0.6254182446876243
0.42666728368852946
0.8892251773882885
0.0846852817974964
0.24565865872547427
0.21859175882720203
0.31122887903048485
0.6216628424078623
0.6790225669472998
0.40384447292807196
0.3467520493377498
0.8366571243962583
0.5694385577535636
0.5055182075546456
0.3002813442260314
0.3502814084315548
0.7701978233506885
0.7001937120233819
0.034311275288159825
0.5210818827802463
0.7882237126705023
0.7247924329317412
0.5187471299394043
0.27473049031692437
0.39271816054563086
0.4385060588281169
0.1774702679120003

0.7149742915826023
0.07447873455209587
0.3358129222716797
0.7053189736316493
0.5828101192080022
0.28201537787551606
0.599805421816116
0.513928465569232
0.4527338297354299
0.3067322107045741
0.37885367260184466
0.6580242622143084
0.1350259218581657
0.485574258237502
0.20570665112101622
0.3948785681805131
0.19439575406176793
0.382323282097079
0.8710335613924639
0.17548526240469986
0.7593179700307807
0.24955953534480513
0.2667906781460408
0.700184325578466
0.6980472463666668
0.7822668130506726
0.3658030817561494
0.7126737253967952
0.7159306002939794
0.6801001071740683
0.4273695804487056
0.6825030177929179
0.41866547001304805
0.7351415355927262
0.4808948986017321
0.21881717303334564
0.8849800651394534
0.4592509399276065
0.8048294571942674
0.2107340603280617
0.6617585088459333
0.2887358236216527
0.7180684476223932
0.6041992411058741
0.2530205162831429
0.37307058632227297
0.778688320426836
0.38737063358650736
0.1249330553724645
0.48363848480197913
0.568366058508734
0.3761718305930524
0.63768

0.3542013109080061
0.8857228306896138
0.8993431691919206
0.6506859222972837
0.4616241137217928
0.7676201289167244
0.574002792030965
0.49217803378228225
0.910953159531609
0.44792692960213576
0.9628238088849479
0.43384077978263175
0.40453417034507017
0.21167142944827988
0.3426222668646604
0.8431979843796494
0.28697365406278
0.5381532069390078
0.7601227776963543
0.7029639145208669
0.7723994772838236
0.6244542415222277
0.7620693513060357
0.4141736696376093
0.5239659290372823
0.8142857711837524
0.4177376229014678
0.03414886327414052
0.1874270025386077
0.7814681087124091
0.28190731419580467
0.6590651555780742
0.5159202985859538
0.59340817584091
0.3736462456896162
0.4206381360807373
0.6635953033194221
0.6856616514048575
0.5419810074772595
0.5102109122523945
0.41313286017468526
0.15811938600195363
0.5712908757635582
0.15838872852582767
0.24436611591250124
0.12172819414696963
0.2151439336324941
0.3084916724247366
0.5029669587326275
0.14387388889106906
0.08649304035914133
0.8118030372043884
0.53

0.425850799722573
0.6410260020843658
0.7179847393204865
0.45147809417065843
0.625188730162511
0.4460934776015336
0.6824215526684345
0.148116096779443
0.23991275454137062
0.7261602754263997
0.9296748987380318
0.2666710166926239
0.566009419866734
0.3488724117958939
0.14147828723267375
0.7804647754681455
0.6216066853644138
0.797523233847255
0.6827069226460103
0.39605371983093723
0.46891618720669465
0.6440906847775556
0.24196715800842705
0.07486370897989889
0.2879683680592159
0.6578594464486583
0.35644989265083804
0.40442956236811156
0.40854697502218457
0.5269762504253381
0.7352236741853181
0.38580664422132915
0.2501580331474994
0.7405518554915343
0.12926628856692304
0.5375049109727624
0.28143828779808844
0.6690358943300915
0.9313856111827866
0.5758534577468879
0.030780605819690724
0.5295394806370872
0.2205076205972961
0.251920767566899
0.5309356117912503
0.7122779119483265
0.44582219639495113
0.2373939063144831
0.7454285405340172
0.47694380488765253
0.4445333544342789
0.762997581075064
0.

0.1534800087363379
0.335465561868996
0.6342255479973488
0.7448663380096994
0.8890779985723922
0.6522421789884446
0.1548247479045502
0.7512785722575864
0.5607789193665487
0.45119907936385023
0.22606362068334693
0.29261115058714054
0.06349848435222348
0.1220599635690408
0.785995462736481
0.38337070108941157
0.4863526281606137
0.6011577608957146
0.25259870647269644
0.2829979360031884
0.4988776401544649
0.8214036819361145
0.2642579334725178
0.7163281384308997
0.25317340890478734
0.7078154836096366
0.3727388997165701
0.579872014520696
0.6778200730804136
0.8517451823012915
0.3809273947109844
0.7568088263455118
0.4876175028594006
0.9432358511047298
0.22880516797217415
0.7049997899494276
0.8666058841491909
0.5695519403891455
0.4197393446623916
0.6487462335937558
0.8410517266450805
0.8084266611606494
0.5959855379668266
0.9663452939440886
0.8515961054259479
0.5636027704657599
0.6011383479221354
0.8615199674328542
0.3235083111447761
0.11377583175729264
0.4505880234380375
0.19497095581173843
0.135

0.40318383004905217
0.5375929084889116
0.3839292324665442
0.4794763877095318
0.261264364512438
0.4480758837585635
0.25945809368234185
0.7148725629370257
0.33307956908227354
0.576907706582926
0.8024342396097126
0.7507797007178493
0.7522610295488212
0.4460588391851783
0.40548485088329783
0.2515859085361061
0.7479710422369726
0.5109182562310894
0.5643229396510504
0.29667473976446523
0.34781519076646106
0.7879596627896144
0.5410207694625387
0.7496371973785199
0.23084129526055644
0.36026181527914347
0.838299037077261
0.5370283711744916
0.281193296937097
0.8333483891120582
0.953926141793621
0.7005914769706398
0.11596272882797508
0.2797199357397038
0.6791482947914731
0.5318539694312108
0.9361319580048162
0.23630176473962855
0.8221733821923998
0.3674762960277641
0.11638339433817033
0.7490010136850582
0.09400244429186021
0.2108504334048711
0.8575342913901774
0.7401860305225201
0.6248977919806068
0.8558909756671459
0.5544530339621914
0.79675146495672
0.4501659973811855
0.34348311661818154
0.6854

0.37465471184059224
0.7977375860548495
0.46046828430891795
0.4084847093126846
0.6030991196483879
0.6304225331845384
0.4696897473690313
0.5268435059248474
0.42838101073470386
0.5315808637536941
0.5909643490338202
0.40303280312447504
0.5164123122126587
0.29248641149556703
0.527775993747655
0.1558284196832623
0.18709940909531683
0.23849539409208148
0.511349018124976
0.3220310606620719
0.909759064081778
0.7831357327447952
0.12812963126087662
0.4620504949857482
0.3067948213999487
0.8110040292890639
0.368782631958146
0.32326223373310703
0.7353344789910672
0.761516323894394
0.21421709965389807
0.3549721556769865
0.8493357172495304
0.16094167659229272
0.07103611470484653
0.41588028032616886
0.1795373847013902
0.019376447015124816
0.4951544559246802
0.9280124313311016
0.2741375351107698
0.5770288725097071
0.6082983759248044
0.48067466516306245
0.40909300502933976
0.6870972923281883
0.6914317104801659
0.39169007626925273
0.48195940455770925
0.6394587766032109
0.47764134081359416
0.57386915602694

0.5730840144876275
0.3811270541586009
0.7919997988483709
0.5473381034953098
0.7544039306306632
0.6595500475453904
0.667419630960268
0.46455430934185915
0.7636561648990859
0.34812627258139156
0.4572968560959979
0.6664860045275555
0.25481202043434487
0.8642193417295484
0.4278132534285658
0.620731402136846
0.7526713421212119
0.2947533474763608
0.8484471001755737
0.48045759911204455
0.3952282663350131
0.6309389554573284
0.289245781772901
0.5862507089933103
0.41326440148179766
0.8458958725222941
0.689777512655508
0.21040541810265626
0.6243250725169595
0.40378086226280274
0.3466012524520962
0.5404266028878036
0.36979872930199137
0.7938499958490609
0.22501058987491118
0.7648886264506667
0.34512865263382114
0.3882658157008607
0.2602396042211659
0.26469351291177307
0.5651755036143663
0.7909878698738134
0.5771093386585562
0.625625188953249
0.2202624109284597
0.6209004734352044
0.528712893516142
0.491791966822292
0.374577574474078
0.5293520297376076
0.6973238025591983
0.8330399679467697
0.2213204

0.39958032904292334
0.9268021727007074
0.06075319644842102
0.8061639602232081
0.32108114651314645
0.6794158837101747
0.39854798259816515
0.7318236101441514
0.16511519742266767
0.09532047842246094
0.6652691903864181
0.6375814398286377
0.6695052880092257
0.39775414587803776
0.4376395349187773
0.10825560164248227
0.38900592896951786
0.33288491552505417
0.3498753081729806
0.44264099167795273
0.6117259420474104
0.7191145857803687
0.22541983453292685
0.2266276017881717
0.42762248328348673
0.2686721739953219
0.8888853001675194
0.4287501798809796
0.3695150486288413
0.41656163261131957
0.31788859568772104
0.7826432424917402
0.9587000845641822
0.8204466783258431
0.8219490184340121
0.7388998087728049
0.2328781279394047
0.3006783324841932
0.5298693659923467
0.40112143791827587
0.2549429363117114
0.7221906666490345
0.6194903399194589
0.4974562625582308
0.20214074540917942
0.43973097779571596
0.6074205325446249
0.8277962435536558
0.5809876969134891
0.6395544369854311
0.5086066949428473
0.37433552096

0.9455698389445795
0.7081981667808437
0.576217890282239
0.07665989417808808
0.39408697373475793
0.46047783951822785
0.6433973906805422
0.4870887964963267
0.20432369791191046
0.531618068250659
0.6691831168709719
0.599816544501637
0.23082594155169844
0.6401840713619386
0.5766709103861599
0.46495551845493643
0.5798011156263345
0.7922719243202778
0.6377006499192901
0.7312440377724428
0.8451551094853945
0.5021625038521857
0.8485299813418733
0.6084203729093342
0.6662952381567805
0.6965563827335242
0.5552035047892958
0.9480748281827275
0.5371040464534508
0.5022330179153272
0.8422720994819841
0.4569231360416617
0.17235830615129488
0.528142968271554
0.48694530444484385
0.8384152261800494
0.6978717219926093
0.3504159392409963
0.23355396373264545
0.5234608687960122
0.8645108606373594
0.20604898156412188
0.17482164692851046
0.6609262309000333
0.7397690130641549
0.6317530389634434
0.20258914294721622
0.21198194016159147
0.8338440335902034
0.9296187952618731
0.8399438673437299
0.3688516108734263
0.2

0.2854165483461095
0.4964423922347796
0.800852465732053
0.7428827067736967
0.28104841504006306
0.3832548932145003
0.4456016527253495
0.0055656848227516025
0.6519305131234288
0.09730905835380733
0.6547154997629168
0.34360674594153706
0.2209357601469875
0.5784932984053903
0.33071058419884164
0.4906841675230962
0.12120436527513137
0.3805856415848942
0.5329090282284173
0.6556740709739411
0.27895881596405786
0.7372578062788889
0.3330936922308076
0.841546867403588
0.7376719363161507
0.5757721095814412
0.47996262305969606
0.5470526156162844
0.5621514511152998
0.5106936905604205
0.5645694287529467
0.5940867211963465
0.786712359535957
0.8542776406190994
0.25724032665629115
0.6272001279286245
0.4069695092299775
0.3404416398391544
0.40698934614185933
0.6209310650523439
0.35818677706439284
0.9612363778562653
0.700631908344487
0.11375922217333115
0.8120546479120878
0.5992526821253485
0.7445592465936457
0.4777385693350232
0.6337604062691607
0.014447520133346792
0.6889256317909163
0.18666438917905645

0.7647766894913851
0.4020950292583285
0.8456474108790946
0.44448060183672644
0.5920136349980725
0.45488208276910785
0.28556238420847385
0.31585418497456325
0.26307642429410305
0.6082534451420882
0.5590853891430925
0.7100315017907728
0.2936769985510577
0.5202166977295912
0.29772069253870714
0.47518524055166395
0.1766807495431138
0.5806812909295862
0.8719490426152301
0.4795427980649799
0.2918858421131143
0.6694595915803446
0.5090271092851955
0.190902147995849
0.4093777582803188
0.5216837864793105
0.7100468390876182
0.8299176720837248
0.06786140574183647
0.3826243651217014
0.34980682688892906
0.7829673358280753
0.7468565394508108
0.39262220071435694
0.8096626476068434
0.5582132885153859
0.565908769040042
0.49655734219648545
0.0678496956594376
0.5509005873320831
0.36144474281153666
0.2986879240113786
0.5023975776162386
0.8376012878325682
0.4903957637362555
0.5897397548856983
0.19808897969634404
0.3956133091761881
0.41677661159247287
0.34421274728626333
0.5141498520402893
0.5239440325159507

0.23180129372549485
0.4414658781196247
0.7761039583236905
0.4834239082579725
0.8789824380581958
0.27116325241527806
0.25023214632550816
0.320329771098361
0.6504520886040388
0.18177354538933552
0.19693643128527236
0.7608870308910669
0.254560426133841
0.19260939193565801
0.8159492404955376
0.25177702296027415
0.29615935033581664
0.6964180864154649
0.5191606612009141
0.806778721191608
0.21868777963576802
0.5819188412571523
0.6690471912499399
0.42362422026898133
0.6043544955423137
0.4078004629694349
0.4346266370323206
0.6468213766875645
0.2354758313461218
0.4224254053078916
0.549848578303735
0.39173594483953705
0.35868821564852565
0.7335876731441492
0.5104049308656153
0.7936894645865955
0.22615265203618448
0.3891853551726848
0.7646793536266063
0.6192225036689657
0.3540352965044644
0.5959918177874859
0.5511358983624859
0.4987061625045436
0.11451765516755619
0.7254416894120336
0.260731291530601
0.35671754892437707
0.41409171040229253
0.44689485461020645
0.7113910663710047
0.3669979188413926


0.4660336398707376
0.2249592500138043
0.7428187740428007
0.2192546149956592
0.12978938641972682
0.25300086521574183
0.370601029600766
0.09353047981214298
0.7808698347166015
0.03335197161705426
0.8003575311762832
0.5085965257794048
0.34873083900156715
0.09082671479090518
0.5669399214091319
0.767568912931992
0.7229027305358181
0.8717829111962855
0.5495266236373864
0.450675387105874
0.492096783529498
0.5708159400880531
0.06136516432994257
0.6167809970130348
0.6280388301476894
0.31263553753105344
0.36018521337410675
0.8035695491139145
0.6339635155279202
0.38194100084204125
0.5652526818737489
0.5121108256249982
0.8313345697689376
0.6411623030120956
0.6902644810331505
0.6213089626693415
0.3995597092686271
0.6565169265465757
0.820747845223022
0.6589665081219735
0.6627421587882691
0.5558319403699349
0.6059020520607458
0.26647129800064245
0.4783003893421277
0.42832210028987383
0.4906492288140347
0.42840400203018403
0.4605116814651381
0.2012633634219588
0.39797911563050425
0.8481089766669966
0.7

0.47973782191413294
0.6330548428917698
0.4648556978936151
0.5694360887594054
0.49698844939514664
0.4782375292685645
0.3986835798109974
0.281793439729033
0.9215220721742444
0.5823479076760149
0.3394600856031403
0.6420127322700628
0.9511254763111514
0.04920237442573447
0.8567980913249836
0.4354139963899619
0.3659801886469882
0.36629425918784625
0.8836491568193178
0.3327079197287222
0.7045428202333021
0.47197171312452446
0.5387899812946318
0.515784392217271
0.4840071353939563
0.5690927000992666
0.8408838007812672
0.4578975108147189
0.6794135190610674
0.7944583090877417
0.330741466940223
0.4614075979914987
0.6375637279856486
0.6510633935817041
0.32195766845376406
0.34340864037870317
0.19172505345978114
0.6641386005686868
0.4641773266576672
0.23230518872347714
0.43005521860748114
0.4701265849748294
0.7156539701152577
0.49844826447590573
0.7886041913716686
0.2582988813558928
0.41883004943745905
0.6811723672408938
0.6168550575575585
0.4901075735150987
0.5125049796901636
0.6834758944686076
0.3

0.9198487945949588
0.3936239715354218
0.754127009391079
0.22014847032007348
0.3330727614823738
0.1963403943196631
0.7267784891718929
0.7316907149420692
0.5064672717800477
0.3772441575168887
0.3903843322098895
0.4059188238609419
0.6297986737290241
0.5341553666212286
0.630712014922652
0.21834763129461798
0.6248536324807494
0.2532148548029532
0.2553422683279978
0.2040639014647556
0.2700214911338915
0.34221472643300793
0.5584792565713084
0.8031023157722272
0.2879794734473715
0.43262004464107967
0.5688484922390965
0.4538686897604419
0.5721633029284743
0.7647969328391029
0.593421485435122
0.706974370679066
0.37356907252257204
0.7927137184826328
0.698386503645295
0.6634266675399042
0.6696324512323347
0.13897322183677957
0.3567680373746315
0.3848427224714605
0.7332931239504501
0.387860881464461
0.6303104006363351
0.24976149638715925
0.8619572860285571
0.2467775298931926
0.37534551286697126
0.3954202096080174
0.8441026888608785
0.5643698174430803
0.6313652988421505
0.627089839015167
0.294620403

0.4866223222265149
0.710419122816727
0.49299945062760775
0.22152843439111042
0.41760693425344864
0.2830336407429605
0.8023814637552807
0.4647529991047873
0.6358731608481968
0.6023826271395037
0.4602036008416436
0.556394510776405
0.4933127818557445
0.7306381666027573
0.6357116695121098
0.7271531485954552
0.6817536223568956
0.4737364107558514
0.7141129775811063
0.42489383842878636
0.656648653278954
0.879046477932869
0.1512454103168192
0.6466735308629011
0.22930160928400223
0.2932214228098362
0.24575382610326163
0.7964106239315308
0.3774313786345399
0.5310067770522564
0.4562519618780365
0.7355683774057026
0.6979811199912227
0.6152109283295731
0.14595990143977344
0.19029272289994328
0.08099940905838132
0.7029234668710077
0.5559913231714033
0.6010563564078227
0.5201876732180314
0.7660384132589659
0.29745401500570723
0.8064028984509828
0.5984600547207202
0.7435878944970105
0.7269408664097463
0.4852050448144319
0.41987114152460175
0.6146968064273352
0.523331292014901
0.736646054309637
0.66970

0.7507460610420917
0.605840959227944
0.09466190909974079
0.8771258045419802
0.45926658937006803
0.8660030602697802
0.307845821549604
0.8357437198209782
0.5249116186182132
0.4463153210284458
0.07870800418569066
0.38932576964923765
0.14581534697895998
0.12088499062493445
0.7392016940483098
0.36841949626117493
0.6105767463620635
0.4010158876990441
0.7316013896826908
0.33124802815495885
0.8362647287666772
0.8969712198528579
0.5613181724345916
0.45232967773921545
0.493827986546787
0.7041900346080396
0.3416813033587608
0.8074740540513362
0.3044562012742183
0.3992363072488563
0.22815217334068777
0.2854769812203986
0.7028345350683214
0.5379484635826792
0.4866741539894714
0.8588758204314696
0.09830497320106833
0.5694046833043764
0.35265371810004736
0.5853490095053657
0.45337873088672487
0.05153808255484469
0.4394141825775224
0.6841957398039624
0.6120075504885836
0.2882654085324516
0.53919916378348
0.7313779583907454
0.6890391125767109
0.3936798153532055
0.6991328103398204
0.40475262294974396
0.

0.7647854358502475
0.14999977629123493
0.9074595937534549
0.5484217434480446
0.5341892522150118
0.3058719805958811
0.9474543347832591
0.6037201796596919
0.8087733912903398
0.44213558250726315
0.6973868148724515
0.48594749707402907
0.27578577525968445
0.8726076182397434
0.7837953312065684
0.6807467353726308
0.44018672941401066
0.46663886221748146
0.30345615843373525
0.5917293743899418
0.10823028875052437
0.691747928859431
0.719475077970595
0.18555626879903445
0.7980422182171342
0.24875185333070085
0.3689983005004135
0.4551500440402459
0.1544372490381007
0.5180106509268602
0.43820870323975253
0.8218277804038028
0.34740964417239706
0.4137261738226797
0.750386388011925
0.6550798707040515
0.3663079978239364
0.6597076240727635
0.15570930963441523
0.47534964756736287
0.49620602931378927
0.7370179010274691
0.6065437908059094
0.717353943816479
0.43127402266040454
0.6745799106340196
0.3781559854962338
0.9279743907284804
0.5179757980068023
0.6828841670980116
0.34301519015882936
0.6300434314731068

0.3642396661364741
0.48547572099083464
0.5532520268552117
0.9618950809746326
0.7800251375552791
0.4428157304022222
0.1461849492556237
0.4081311845550377
0.2455855563384783
0.38229440726647784
0.6363034563669472
0.6740728316343331
0.3159518150420039
0.5288013456951259
0.3292505198189015
0.6156923840892037
0.14965707276522014
0.312436173197004
0.12919847362358658
0.42218422703141567
0.738694741674709
0.1689681174702554
0.8771381438356017
0.6452798012656753
0.28836768761511444
0.7464322471225735
0.23886641995594435
0.6526672447830058
0.9696281138041852
0.5634835563957201
0.7320183958614762
0.4433895498940685
0.5478646363088218
0.47996714157626474
0.7782476988391797
0.185664944566044
0.482851040987301
0.657344917315936
0.29750613266906645
0.1644263009736325
0.4912736334303154
0.6507738770524446
0.308316086082679
0.3926381593516653
0.6962753751364735
0.6061226733567805
0.4122086120043092
0.7010113937370182
0.5647082250325199
0.49144629689776237
0.7915024300684618
0.5178515974060829
0.813166

0.3106641863971249
0.15752785483102408
0.3211664425640963
0.29325294636125027
0.10264497127155368
0.8077472609439639
0.5702655620397914
0.03529584162498994
0.5640242072128304
0.7229445067512928
0.11946848231132379
0.5083343590186534
0.7399667230313285
0.5530432248391959
0.7400292705842265
0.3841738720281479
0.34648647474147237
0.14566652759560741
0.6169406448754085
0.7279814297680219
0.8539633136228122
0.7683810897112523
0.6067683812339822
0.7659829573099424
0.700440406092653
0.2338861247454897
0.2636325905037701
0.6062889981753157
0.7200422003340401
0.6603845308415146
0.6780345971901868
0.544152444382182
0.2956133582866498
0.6257304061912311
0.6665027028049978
0.39036892348965824
0.6354508690967783
0.4204521371313724
0.32821527179630783
0.44879999037714424
0.568115413216236
0.49549041327075793
0.18844280734061436
0.8307976190904501
0.5224595859486104
0.17002335282771397
0.440295235142727
0.5226486345332233
0.8301062061193488
0.5066697431237953
0.6156075473128002
0.11659200080994417
0.

0.8807504396298109
0.8088946136030132
0.3248658856955907
0.346488406835508
0.6361137819730689
0.531931958281617
0.2827160030995365
0.5195449009857703
0.2597756572841333
0.44828691061853
0.4206894845953776
0.7996020009461764
0.634421363070638
0.7824835585423464
0.19136362779086052
0.6602344538512587
0.555856915375925
0.2418979008305377
0.7618541235616744
0.38666226697210543
0.23021135934302983
0.4639534874502632
0.7816993901274405
0.5821604434696885
0.4747916768079661
0.5786736846423619
0.7377426150443863
0.29462185453016804
0.6072308253084631
0.4937041111442512
0.1117479810390214
0.14132040356192088
0.7401952587397085
0.21434339663106158
0.5733339734153188
0.43984350160912394
0.7432702772910382
0.21008752543786374
0.7427435143246647
0.6447454391539036
0.8957206271932269
0.5762952784510104
0.45288112679650694
0.6524707425210928
0.8234061158396007
0.8688149529512255
0.3671305230082934
0.6127645690156784
0.3937577598151123
0.5218213970612273
0.34148552892271633
0.3316033335001939
0.321785

0.6630148357801122
0.5354433393402465
0.31299541988407337
0.8924064842033835
0.09191520633950231
0.45117489003242306
0.7211510239828675
0.9126831280920551
0.37765944711876887
0.38919975587245625
0.9017658197013636
0.37278536272570795
0.07949966038491801
0.35931518196950285
0.9535948245199487
0.4507379307625799
0.4119149730030422
0.2498847706177023
0.6610757598768773
0.4396297205571487
0.5177432243977491
0.27054600040150495
0.4566831835787782
0.49760932053476536
0.6727885211645186
0.14338865022743377
0.4144233344810113
0.8828023175002365
0.8815807177376486
0.263392427041884
0.09428492201136628
0.5891661897702195
0.7481735243471748
0.6921723732712399
0.519412427414922
0.4348918458469069
0.9209701551753039
0.5225022122758879
0.7281786092822432
0.7159916504980242
0.5897762401082379
0.2785835571073408
0.22518115271661657
0.649900367705664
0.3530277094006086
0.7213359918860476
0.7276280021991071
0.7943402013091401
0.49265374416219376
0.5022235180411462
0.43102791259140427
0.7278336771191208


0.3012257605649556
0.6869858365848821
0.3553749476546555
0.07885695270279396
0.14337206913975012
0.8806761694157499
0.4220622724011306
0.17228248577293037
0.10890824265549161
0.3878062938315486
0.3430498052902885
0.8751802352148879
0.579080502277683
0.5694337629111481
0.6405425454522505
0.7294178027573893
0.2690242938416431
0.5435890964293576
0.7301292763052779
0.8178892798260347
0.8033725685968061
0.5024874326967836
0.6904504214063153
0.658815621474042
0.9681890033088639
0.4592809658008525
0.5851785198763385
0.9016501124566653
0.4079476008917256
0.6949503728690789
0.41329081066319123
0.5287333138748075
0.7914982500736311
0.641124594863835
0.1448159860818488
0.4398809039849956
0.666987883579767
0.3255142830188318
0.4325238283852857
0.6357737178325528
0.6799307405970698
0.8589942016698946
0.20634299372454426
0.780787647913101
0.7090699433449164
0.3340102286005176
0.41374011134525385
0.7392705982964233
0.7176913984794775
0.5446227096008074
0.20223523787814887
0.3902384825877919
0.6240298

0.38456856231185527
0.568002274326891
0.518967270028365
0.6509566674137174
0.8113776341488794
0.21926416353974706
0.48392405694881835
0.50203208933244
0.9018268845581867
0.4061952713637023
0.1464238736942739
0.7151334732952112
0.24051496806358927
0.5940040008176404
0.2502687012720748
0.15664228794796659
0.3919925672031776
0.416322821688924
0.6680430828858203
0.7081276553770631
0.357197191994841
0.2697772803498965
0.7530612630721517
0.4983324759668294
0.5537444114198409
0.8328235585528652
0.7101265247273062
0.2640304649866843
0.716633606044134
0.6367905175211998
0.7374823666675338
0.7319856826075687
0.6489096375888618
0.179270953119934
0.21696135590788088
0.7962833697063215
0.713150076776459
0.5260827820599573
0.4904800517818956
0.631275360768546
0.4845285119848496
0.47852761703451663
0.439879751250756
0.3417274559766069
0.28115946174696005
0.6999219982932363
0.6819187083947679
0.36320364291899143
0.4684199387251039
0.5624906606356173
0.18810006351112507
0.5276345781061161
0.48036292295

0.8763485686815112
0.8877006292731855
0.7832192618258581
0.7041275413956128
0.40251697979542966
0.887704241147089
0.647213305747573
0.859438164797089
0.845924261833256
0.3495792563716056
0.4479325188169248
0.5337857014973355
0.4359185512331221
0.19494528853099624
0.6719297364734005
0.692912619409393
0.8687280068592521
0.5741127348621363
0.6502479318514661
0.7147046593303225
0.9300094695889288
0.689603283911452
0.8945003818980476
0.525613079295831
0.3911235913249208
0.29539538008905347
0.08517762251819083
0.642260263115669
0.4390849551765285
0.4645017359877127
0.5981354504025586
0.47547307944430556
0.8039010661634394
0.5385867738083411
0.7245232926759168
0.26271376202665486
0.5732583094963329
0.14045286170791246
0.36564903616441863
0.5038492882294957
0.16842461981573145
0.5510155682176151
0.525803165392815
0.4145904840501634
0.7444415055099555
0.5669739942153895
0.8494727004205193
0.4214745126796896
0.3398464055323615
0.561186782239766
0.3972410665909984
0.5185212494291488
0.11752442908

0.6312821966687037
0.6699264190355417
0.8537827649552546
0.3500386044250422
0.49096797322848884
0.5096326057266434
0.22786182189334503
0.6335418027576357
0.8101771706168314
0.2908106029297078
0.4650910135068965
0.41619911037836366
0.321883771855643
0.6586791676501985
0.811878329587707
0.8458725612454019
0.545185902820311
0.6970999903237965
0.6773558720317925
0.8094827983284804
0.8142513623165806
0.7657755869964531
0.8163838070878132
0.5049398344760068
0.565921680802592
0.5787920355821508
0.3148052446555941
0.28831380441358234
0.8090664376463278
0.42239221298482066
0.5659535091689484
0.7118077991345417
0.399737133600989
0.9198135968552744
0.9527714096369196
0.9792485886686902
0.3594055410676891
0.3908566879010138
0.7566591753667121
0.5798946649936016
0.32231047620640063
0.2418525020549025
0.5883746355296621
0.731954862781647
0.5081692740491346
0.43401764516053165
0.2200851125015836
0.760231533072204
0.6376886812265908
0.7394796547271449
0.9229422650107056
0.6575192816183406
0.7678875042

0.3221073098661499
0.7403750106386301
0.7437379624837621
0.6486489569251642
0.3469162410417536
0.33382571543331524
0.5297660991802682
0.4112587310620644
0.6325503668355041
0.07318915660029114
0.8874610575130807
0.32167599475063385
0.5695927382790343
0.5526247239022726
0.5866665112563976
0.16936086878565657
0.44642649025795295
0.6315985437080853
0.33811752699786823
0.4060388491225201
0.5369864453108025
0.29360043595993757
0.41228331795469175
0.4099859716795111
0.7144836905476366
0.31418947961955274
0.1799383956811495
0.45731762206560506
0.6013509024470428
0.7667183965341758
0.8046556558101627
0.5621891799569684
0.5474544268628717
0.4448582027661109
0.6999335343681631
0.29582958109832636
0.644673021175748
0.943501519061109
0.6002126417540865
0.36343283070578936
0.5829469792337495
0.3694472513377672
0.41592793887414414
0.5929637863692472
0.910088869733086
0.861533785314567
0.6177202695830724
0.62106617857538
0.6731924343874731
0.09384653010118718
0.7769550137890789
0.48141992320140853
0.3

In [18]:
print('min: {} \nmax: {}'.format(min(target_pred), max(target_pred)))

min: 0.00132093576194957 
max: 0.9987355308409461


Min: 0.00132093576194957

Max: 0.9987355308409461